router/connection.go

```diff
@@ -292,7 +309,13 @@ func (conn *LocalConnection) run(actionChan <-chan ConnectionAction, finished ch
 		return
 	}
 
-	conn.Log("completed handshake; using protocol version", conn.version)
+	if conn.SessionKey == nil {
+		conn.Decryptor = NewNonDecryptor()
+	} else {
+		conn.Decryptor = NewNaClDecryptor(conn.SessionKey, conn.outbound)
+	}
```

Comment:
I think this should be done as part of `DoIntro`, in `setup[No]Crypto`. That way all the crypto setup is in one place. It also means we can remove `conn.SessionKey`.